lib/plug/parsers.ex

```diff
@@ -236,4 +241,9 @@ defmodule Plug.Parsers do
       raise UnsupportedMediaTypeError, media_type: "#{type}/#{subtype}"
     end
   end
+
+  defp merge_path_params(%{path_params: %Plug.Conn.Unfetched{}} = conn), do: conn
```

Comment:
Yes, it looks like the query/body params merging can be removed from `reduce/6` if it doesn't have to return anymore. Instead that can happen as a final step.